### DISCUSSION TITLE
Document.convert_to_pdf() Removes Transparency in PDFs

### DISCUSSION THREAD

**Discussion by Danishkhilji** [pymupdf/PyMuPDF#4101] [Looking for help]
I'm trying to use the Document.convert_to_pdf() method from PyMuPDF (fitz) to manipulate a PDF document that contains both vector graphics and images with transparency. However, when using this method on PDFs containing transparent elements, the transparency is removed, and the result is different from what I expect.

**Code Example:**
```
def get_design_layer(input_pdf: BytesIO, visible_rect: list) -> BytesIO:
    try:
        input_pdf.seek(0)        
        doc = fitz.open(stream=input_pdf, filetype="pdf")        
        page = doc[0]
        
        # Get layer UI configurations
        ui_configs = doc.layer_ui_configs()
        design_layer_index = None
        
        # Find the "Design" layer
        for i, config in enumerate(ui_configs):
            if "Design" in config["text"]:
                design_layer_index = i
                break

        if design_layer_index is None:
            raise ValueError("Design layer not found!")

        # Turn off all layers
        for i in range(len(ui_configs)):
            doc.set_layer_ui_config(i, action=2)  # Turn off
        
        # Turn on the "Design" layer
        doc.set_layer_ui_config(design_layer_index, action=1)  

        pdf_bytes = doc.convert_to_pdf(from_page=0, to_page=-1)

        return BytesIO(pdf_bytes)
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"An error occurred: {str(e)}")

```

**Problem:**

- The code works fine when applied to a simple PDF containing vector graphics.
- However, when I try the same on a PDF that includes both vector graphics and images (where some images and vectors have transparency), the resulting PDF removes all transparency.
- Not only removes  transparency also change clipping path position

**Example:**

- Original PDF (Expected result): 
![image](https://github.com/user-attachments/assets/6967ecb6-4300-4e4d-8cc8-1ef1ca017557)

- Resulting PDF (Actual result): [Image 2] (transparency removed and also change position)
![image](https://github.com/user-attachments/assets/0854ea9e-abd6-40ac-80e1-d11afa151623)

following error i am getting on terminal

```
MuPDF error: syntax error: cannot find ExtGState res`
ource 'BlendMode0'

MuPDF error: syntax error: cannot find ExtGState resource 'BlendMode0'

MuPDF error: syntax error: cannot find ExtGState resource 'BlendMode1'

MuPDF error: syntax error: cannot find ExtGState resource 'BlendMode0'

MuPDF error: syntax error: cannot find ExtGState resource 'BlendMode0'

MuPDF error: syntax error: cannot find ExtGState resource 'BlendMode1'
..... more times same

invalid marked content and clip nesting
... repeated 4 times...
syntax error: cannot find ExtGState resource 'BlendMode0'
encountered syntax errors; page may not be correct
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode1'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode1'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode1'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode1'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode0'
syntax error: cannot find ExtGState resource 'BlendMode1'
syntax error: cannot find ExtGState resource 'BlendMode0'
encountered syntax errors; page may not be correct

```

**Expected behavior:**

- I expect the transparency and clipping path in both the images and vectors to be preserved in the resulting PDF.

here is pdf file:
[test-file.pdf](https://github.com/user-attachments/files/17976634/test-file.pdf)

**Comment by Danishkhilji** [comment]
Hi @JorjMcKie ,

Thanks for all your help! If you have a moment, could you take a look at my latest issue regarding transparency in PDFs with convert_to_pdf()?

Appreciate it!

**Reply by Danishkhilji** [reply]
@JorjMcKie

**Comment by JorjMcKie** [comment]
Sorry, I swamped with work lately and cannot deal with your issue.
Hope for your understanding.
